### PR TITLE
Fix garbage collection about t.droppedSeries in QueueManager.SeriesReset.

### PR DESCRIPTION
**Pull request by kevinForMyself**
The droppedSeries in QueueManager struct is use for ref the Series which will be droped by config remote action.
The key(ref) of droppedSeries map will never insert into the seriesSegmentIndexes:
func (t *QueueManager) StoreSeries(series []record.RefSeries, index int) {
    // ... ...
    for _, s := range series {
        ls := processExternalLabels(s.Labels, t.externalLabels)
        lbls := relabel.Process(ls, t.relabelConfigs...)
        if len(lbls) == 0 {
            t.droppedSeries[s.Ref] = struct{}{}
            **continue                                   // the Ref of drop Series is never insert into  t.seriesSegmentIndexes**
        }
        t.seriesSegmentIndexes[s.Ref] = index
        t.internLabels(lbls)
        // ... ....
    }
}
So, garbage collection for t.droppedSeries doesn't work. 
Because the s.Ref of the droppedSeries and seriesSegmentIndexes are  different part.

### CODE REVIEW

File: storage/remote/queue_manager.go [hunk 1]
```diff
@@ -541,7 +541,7 @@ func (t *QueueManager) StoreSeries(series []record.RefSeries, index int) {
 		ls := processExternalLabels(s.Labels, t.externalLabels)
 		lbls := relabel.Process(ls, t.relabelConfigs...)
 		if len(lbls) == 0 {
-			t.droppedSeries[s.Ref] = struct{}{}
+			t.droppedSeries[s.Ref] = index
 			continue
 		}
 		t.seriesSegmentIndexes[s.Ref] = index
```
Comment (csmarchbanks) [inline storage/remote/queue_manager.go:547]: The idea of `seriesSegmentIndexes` is that it keeps track of all refs -> index mappings. I think all that needs to be done in this PR is move 
```
t.seriesSegmentIndexes[s.Ref] = index
```
to the beginning of the `for` loop. Then the loop in `SeriesReset` will also properly work without needing to change `droppedSeries` to an int from a struct. It is probably worth adding a comment as well.

**Review by csmarchbanks — COMMENTED** [review]
Thanks, and nice find!

Would it be possible to add an assertion to `TestSeriesReset` as well?

File: storage/remote/queue_manager.go (hunk 1 above)
Comment (kevinForMyself) [reply storage/remote/queue_manager.go:547]: Sure, I will make the other PL to fix this one. Thanks for your advice.

**Review by csmarchbanks — APPROVED** [review]

**Comment by csmarchbanks** [comment]
Thanks!

**Pull request merged**